core/src/main/java/org/infinispan/reactive/publisher/impl/MaybeValueRetainedFunction.java

```diff
@@ -0,0 +1,12 @@
+package org.infinispan.reactive.publisher.impl;
+
+import java.util.function.Function;
+
+public interface MaybeValueRetainedFunction<I, O> extends Function<I, O> {
```

Comment:
I think if we flip the way of thinking about this interface, it might be easier to understand. For example if we call it `ModifiedValueFunction` and the getter `isModifiied`, WDYT?

So `retainsOriginalValue()` becomes `!isModified()`.

Reply:
Sure, I was hoping that someone would comment on this interface, cause I couldn't think of a good name :)

I will also add some Javadoc to the interface.